packages/fx-core/src/common/spec-parser/utils.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import axios from "axios";
+import fs from "fs-extra";
+import { ConstantString } from "./constants";
+import { OpenAPIV3 } from "openapi-types";
+
+export async function isYamlSpecFile(specPath: string): Promise<boolean> {
+  const isRemoteFile = specPath.startsWith("http:") || specPath.startsWith("https:");
+  const fileContent = isRemoteFile
+    ? (await axios.get(specPath)).data
+    : await fs.readFile(specPath, "utf-8");
+
+  try {
+    JSON.parse(fileContent);
+    return false;
+  } catch (error) {
+    return true;
+  }
+}
+
+export function isSupportedApi(method: string, path: string, spec: OpenAPIV3.Document): boolean {
+  debugger;
+  const pathObj = spec.paths[path];
+  method = method.toLocaleLowerCase();
+  if (pathObj) {
+    if (method === ConstantString.GetMethod && !pathObj[method]?.security) {
+      const operationObject = pathObj[method] as OpenAPIV3.OperationObject;
+      if (operationObject.parameters?.length === 1) {
```

Comment:
Is there the case for zero parameter? E.g. list all may not require param.

Reply:
![image](https://github.com/OfficeDev/TeamsFx/assets/5545529/b88d31a1-7123-4f81-8d7c-c381ccb874bd)

Thanks, from the design doc, it said that only 1 paramter, however, I think your suggestions is correct, we should also support 0 parameter

Reply:
Updated